### PR TITLE
update cartpole for 0.3

### PR DESCRIPTION
**Pull request by brettkoonce**
@rxwei okay, the new problem is that there are models both in stable and then new dev has been done against master.  how do you want to get things in sync?  happy to redo my pr's if that is simpler.  otherwise would be better to do a sync before doing significant cleanup that will have to be rebased later.

### CODE REVIEW

File: Gym/CartPole.swift [hunk 1]
```diff
@@ -181,9 +180,8 @@ while true {
       episodes: episodes, actionCount: actionCount)
 
     let gradients = gradient(at: net) { model -> Tensor<Float> in
-        let context = Context(learningPhase: .training)
-        let logits = model.applied(to: input, in: context)
-        let loss = softmaxCrossEntropy(logits: logits, oneHotLabels: target)
+        let logits = model.applied(to: input)
+        let loss = softmaxCrossEntropy(logits: logits, probabilities: target)
```
Comment (brettkoonce) [inline Gym/CartPole.swift]: also a note here that this keyword changed for reference

File: Gym/CartPole.swift [hunk 2]
```diff
@@ -181,9 +180,8 @@ while true {
       episodes: episodes, actionCount: actionCount)
 
     let gradients = gradient(at: net) { model -> Tensor<Float> in
-        let context = Context(learningPhase: .training)
```
Comment (rxwei) [inline Gym/CartPole.swift]: Outside the `while true` block, you need to add a:
```swift
Context.local.learningPhase = .training
```

**Comment by rxwei** [comment]
It would be great to pull in all changes from stable to master now and lock down stable. When v0.3 gets released, we then push master to stable.

**Comment by brettkoonce** [comment]
added flag, see #97 for stable rebase!

File: Gym/CartPole.swift (hunk 2 above)
Comment (brettkoonce) [reply Gym/CartPole.swift]: added at L174, not L131, right?

Comment (rxwei) [reply Gym/CartPole.swift]: right

**Comment by rxwei** [comment]
Thanks so much!

**Pull request merged**